subsys/bluetooth/host/hci_core.c

```diff
@@ -4643,13 +4649,10 @@ int bt_disable(void)
 {
 	int err;
 
-	if (atomic_test_and_set_bit(bt_dev.flags, BT_DEV_DISABLE)) {
+	if (!atomic_test_and_clear_bit(bt_dev.flags, BT_DEV_READY)) {
```

Comment:
This change does open up for a race condition between `bt_enable` and `bt_disable`, as `bt_enable` can now be be executed before `bt_disable` returns, but I guess that this is what you referred to an application bug?

Reply:
In my opinion we should try to have the extra safety against race conditions.

I'm not fully aware of what use cases `bt_disable` is for, but if it's a kind of power-save then I can see an application that is in progress of disabling suddenly need to re-enable it because of e.g. user interaction. We should try to prevent undefined behavior if the application forgets to manage this with a lock.